### PR TITLE
UTOPIA-1207 Add afterScenario and refactor code

### PR DESCRIPTION
**Pull request by akhilfw**
# Description

This PR includes the following proposed change(s):

- Add afterScenario and refactor code

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/test/features/featurefiles/sanity/login.feature [hunk 1]
```diff
@@ -5,20 +5,18 @@ Feature: Log in to DPIA
...
     Then I am able to verify title
-    And I sign out successfully
 
     Examples:
       | username | role    |
       | DORGALE  | MPO     |
       | DOND     | Drafter |
       | ANDYA    | CPO     |
-
+
```
Comment (kushal-arora-fw) [inline src/frontend/test/features/featurefiles/sanity/login.feature:14]: unnecessary spaces.

File: src/frontend/test/features/featurefiles/sanity/login.feature [hunk 2]
```diff
@@ -64,4 +61,15 @@ Feature: Log in to DPIA
...
+    Given I am on the login page
+    When I login with user <username> having role <role>
+    Then I sign out successfully
 
+    Examples:
+      | username | role    |
+      | DORGALE  | MPO     |
+      | DOND     | Drafter |
+      | ANDYA    | CPO     |
```
Comment (kushal-arora-fw) [inline src/frontend/test/features/featurefiles/sanity/login.feature:75]: Please add EOL - [an Empty blank line]

**Review by kushal-arora-fw — APPROVED** [review]
a couple of nits changes added.

File: src/frontend/test/features/featurefiles/sanity/login.feature (hunk 2 above)
Comment (akhilfw) [reply src/frontend/test/features/featurefiles/sanity/login.feature]: Thanks a lot @kushal-arora-fw. Added new line to Login.feature and removed unnecessary spaces

**Review by kushal-arora-fw — APPROVED** [review]
Those are still unfixed.. but very minor.. feel free to merge.. and we can connect in the morning and I can explain how the requested comments are still unresolved.

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
[Reversing approval] Docker Build failing. Please have a look.

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**